### PR TITLE
Catch os... errors

### PR DESCRIPTION
**Pull request by Bwko**

### CODE REVIEW

File: cmd/dump.go [hunk 1]
```diff
@@ -79,7 +79,10 @@ func runDump(ctx *cli.Context) error {
 	log.Printf("Packing dump files...")
 	z, err := zip.Create(fileName)
 	if err != nil {
-		os.Remove(fileName)
+		osErr := os.Remove(fileName)
```
Comment (tboerger) [inline cmd/dump.go]: Maybe we should keep ignoring this error?

File: cmd/dump.go [hunk 2]
```diff
@@ -102,7 +105,10 @@ func runDump(ctx *cli.Context) error {
 	}
 	// FIXME: SSH key file.
 	if err = z.Close(); err != nil {
-		os.Remove(fileName)
+		osErr := os.Remove(fileName)
```
Comment (tboerger) [inline cmd/dump.go]: Same here

File: cmd/serve.go [hunk 3]
```diff
@@ -13,6 +13,8 @@ import (
 	"strings"
 	"time"
 
+	goLog "log"
```
Comment (tboerger) [inline cmd/serve.go]: Do we already have a logger here?

File: cmd/web.go [hunk 4]
```diff
@@ -654,8 +655,10 @@ func runWeb(ctx *cli.Context) error {
 	case setting.FCGI:
 		err = fcgi.Serve(nil, m)
 	case setting.UnixSocket:
-		os.Remove(listenAddr)
-
+		osErr := os.Remove(listenAddr)
+		if osErr != nil {
+			osLog.Fatalf("Fail to remove unix socket directory %s: %v", listenAddr, osErr)
```
Comment (tboerger) [inline cmd/web.go]: Just use the already available logger?

File: main.go [hunk 5]
```diff
@@ -37,5 +38,7 @@ func main() {
 		cmd.CmdAdmin,
 	}
 	app.Flags = append(app.Flags, []cli.Flag{}...)
-	app.Run(os.Args)
+	err := app.Run(os.Args)
```
Comment (tboerger) [inline main.go:42]: Generally I'm wrapping this directly into println

File: models/pull.go [hunk 6]
```diff
@@ -622,8 +626,11 @@ func (pr *PullRequest) PushToBaseRepo() (err error) {
 	headFile := fmt.Sprintf("refs/pull/%d/head", pr.Index)
 
 	// Remove head in case there is a conflict.
-	os.Remove(path.Join(pr.BaseRepo.RepoPath(), headFile))
-
+	file := path.Join(pr.BaseRepo.RepoPath(), headFile)
+	err = os.Remove(file)
+	if err != nil {
+		return fmt.Errorf("Fail to removee dir %s: %v", path.Join(pr.BaseRepo.RepoPath(), headFile), err)
```
Comment (tboerger) [inline models/pull.go]: Typo

File: cmd/dump.go (hunk 1 above)
Comment (lunny) [reply cmd/dump.go]: `_ := os.Remove(fileName)` ?

File: main.go (hunk 5 above)
Comment (lunny) [reply main.go:42]: @tboerger Don't do that. Last time, it results in a bug I spent many time to fix it.

Comment (tboerger) [reply main.go:42]: But now it's not a big difference

Comment (tboerger) [reply main.go:42]: It's not using any logger, it's only printing to stdout. So we will have the same bug again

Comment (lunny) [reply main.go:42]: yes, don't output anything to stdout when git via ssh.

File: cmd/dump.go (hunk 1 above)
Comment (bkcsoft) [reply cmd/dump.go]: Here's a dump question... Why is it deleting a file it can't create in the first place? :trollface:

File: cmd/serve.go [hunk 7]
```diff
@@ -51,7 +53,10 @@ func setup(logPath string) {
 
 	if setting.UseSQLite3 || setting.UseTiDB {
 		workDir, _ := setting.WorkDir()
-		os.Chdir(workDir)
+		osErr := os.Chdir(workDir)
+		if osErr != nil {
+			goLog.Fatalf("Fail to change directory %s: %v", workDir, osErr)
```
Comment (bkcsoft) [inline cmd/serve.go]: Printing server-internal dirs to any random client?! 😱

**Review by bkcsoft — CHANGES_REQUESTED** [review]
Don't do this 😆

**Comment by bkcsoft** [comment]
Overall, change the format to this instead whenever possible. Keeps errors(variables) bound to their context(block).
```
if err := os.Remove(...); err != nil {
  [...]
}
```

File: cmd/dump.go (hunk 1 above)
Comment (tboerger) [reply cmd/dump.go]: Good catch :D

File: main.go (hunk 5 above)
Comment (Bwko) [reply main.go:42]: @lunny how do you want me to fix this?

File: cmd/serve.go (hunk 3 above)
Comment (lunny) [reply cmd/serve.go]: ^

File: main.go [hunk 8]
```diff
@@ -37,5 +38,7 @@ func main() {
...
+	fmt.Printf("Fail to run app with %s: %v", os.Args, err)
```
Comment (lunny) [inline main.go]:
```Go
if err != nil
```

File: modules/log/log.go [hunk 9]
```diff
@@ -6,6 +6,7 @@ package log
 
 import (
 	"fmt"
+	"log"
```
Comment (lunny) [inline modules/log/log.go]: module/log

File: modules/ssh/ssh.go [hunk 10]
```diff
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"io"
 	"io/ioutil"
+	osLog "log"
```
Comment (lunny) [inline modules/ssh/ssh.go]: same

**Review by lunny — CHANGES_REQUESTED** [review]

File: cmd/dump.go (hunk 1 above)
Comment (strk) [reply cmd/dump.go]: Maybe zip.Create might fail some some after file creation?

**Comment by Bwko** [comment]
I've updated my PR.

**Comment by lunny** [comment]
LGTM

**Review by lunny — APPROVED** [review]

File: main.go (hunk 5 above)
Comment (bkcsoft) [reply main.go:42]: @lunny `cmd/serve.go` always returns `nil`, so this isn't really an issue here :)

**Review by bkcsoft — APPROVED** [review]

**Comment by bkcsoft** [comment]
LGTM

**Comment by tboerger** [comment]
This partially solves https://github.com/go-gitea/gitea/issues/257

**Pull request merged**